### ISSUE TITLE
Spyder attempts to set attributes of timedelta objects when opened in Variable Explorer, resulting in an AttributeError: Readonly attribute 

### ISSUE THREAD

**Issue by CAM-Gerlach** [spyder-ide/spyder#5999]
## Description

When opening/"editing" timedelta objects in Spyder's variable explorer, one is able to doubleclick an attribute (e.g. days, seconds, microseconds, etc) to edit it and insert a different value. However, upon hitting enter or deselecting the field, the following error occurs:

```
>>> Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\spyder\widgets\variableexplorer\collectionseditor.py", line 623, in setModelData
    self.set_value(index, value)
  File "C:\Anaconda3\lib\site-packages\spyder\widgets\variableexplorer\collectionseditor.py", line 387, in set_value
    index.model().set_value(index, value)
  File "C:\Anaconda3\lib\site-packages\spyder\widgets\variableexplorer\collectionseditor.py", line 342, in set_value
    self._data[ self.keys[index.row()] ] = value
  File "C:\Anaconda3\lib\site-packages\spyder\widgets\variableexplorer\collectionseditor.py", line 83, in __setitem__
    setattr(self.__obj__, key, value)
AttributeError: readonly attribute
```

This could have something to do with the apparent incomplete support for timedeltas in variable explorer; for example, `value_to_display()` in utils explicitly checks for them to display in an elseif along with other supported types, but elsewhere they are apparently categorized as an unsupported type (or not included in the list of supported types), and thus not shown when "hide unsupported types" is selected, and no actual representation of them is shown using that function, but rather the same boilerplate as the other unsupported types (although the code exists to do so, it is not called).

A short term patch shouldn't be too complex to derive, but a better solution would just be to properly implement full support for timedeltas which shouldn't be terribly complex; I can take a stab at it now.

**What steps will reproduce the problem?**

1. Create a named timedelta object
2. Open it in in a variable explorer (technically object explorer) subwindow (e.g. doubleclick or right click to edit)
3. Attempt to change any value therein

**What is the expected output? What do you see instead?**

Expected: Either editing such values is disallowed by setting and respecting the readonly property; they can be edited in the object explorer window without error, or (ideally) they can be edited directly in the variable explorer as can `datetime.date`s and `datetime.datetime`s, depending on what is most appropriate.

Main obstacle I would think is finding an appropriate Qt editor class to handle timedeltas, or hacking it in/converting it to and from the existing QDateTimeEditor, etc. class. Alternatively, for now perhaps we could just make the existing object explorer dialog allow editing of the relevant attributes, if possible.

## Version and main components

* Spyder Version: 3.2.4 
* Python Version: 3.6.3
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.3 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26.1 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.4 (OK)

```

**Comment by CAM-Gerlach** [comment]
This also it may be a more general issue with attempting to edit values for arbitrary unsupported objects when the former are read-only/immutable attributes, but in those cases (at least e.g. with sets), I instead get a slightly different error for the last line, e.g. `AttributeError: attribute '__doc__' of 'builtin_function_or_method' objects is not writable`. I suppose this sort of thing can either be handled gracefully, or just wontfixed given the issue is with the arbitrary object type being unsupported, not with Spyder directly.

**Comment by CAM-Gerlach** [comment]
Welp, my Google-fu failed me, and after wondering why it didn't work already given the code, I figured out it was already fixed in 3.2.5. Silly me!
